Application/EdFi.Ods.Api/Security/Authorization/Repositories/AggregateRootCriteriaProviderDecoratorBase.cs

```diff
@@ -54,80 +60,213 @@ public ICriteria GetCriteriaQuery(TEntity specification, IQueryParameters queryP
+                    }
+
+                    // Invoke the filter applicators against the current query
+                    foreach (var applicator in applicators)
+                    {
+                        // Make claim values available under various names used by different contexts (NOTE: long term goal would be to eliminate this variation) 
+                        var parameterValues = new Dictionary<string, object>
+                        {
+                            { "SourceEducationOrganizationId", filterDetails.ClaimValues },
+                            { filterDetails.ClaimEndpointName, filterDetails.ClaimValues },
+                        };
+
+                        // Apply the authorization strategy filter
+                        applicator( criteria, disjunction, parameterValues, joinType);
+
+                        filterApplied = true;
+                    }
+                }
+
+                if (filterApplied)
+                {
+                    conjunction.Add(disjunction);
+                }
+
+                return filterApplied;
+            }
+
+            void ApplyJunctionsToCriteriaQuery()
+            {
+                if (disjunctionFiltersWereApplied)
+                {
+                    if (conjunctionFiltersWereApplied)
+                    {
+                        mainConjunction.Add(mainDisjunction);
+                    }
+                    else
+                    {
+                        criteria.Add(mainDisjunction);
+                    }
+                }
+
+                if (conjunctionFiltersWereApplied)
+                {
+                    criteria.Add(mainConjunction);
+                }
+            }
+
+            void EnsureRequestAuthorized()
+            {
+                if (unsupportedAuthorizationFilters.Any() && !conjunctionFiltersWereApplied && !disjunctionFiltersWereApplied)
                 {
                     if (_logger.IsDebugEnabled)
                     {
-                        _logger.Debug($"Unable to authorize access to '{typeof(TEntity).FullName}' because none of the following authorization filters were defined: '{string.Join($"', '", unsupportedAuthorizationFilters)}'.");
+                        _logger.Debug(
+                            $"Unable to authorize access to '{typeof(TEntity).FullName}' because none of the following authorization filters were defined: '{string.Join($"', '", unsupportedAuthorizationFilters)}'.");
                     }
 
+                    string[] distinctClaimEndpointNames = authorizationFiltering
+                        .SelectMany(s => s.Filters.Select(f => f.ClaimEndpointName))
+                        .Distinct()
+                        .OrderBy(x => x)
+                        .ToArray();
+
+                    // TODO: Multiple authorization strategy support -- is there a better way to message the error (which used to combine claim endpoints from a single auth strategy)?
```

Comment:
Should a ticket be created in tracker for this? Or is this TODO already being handled as part of a subsequent ticket?